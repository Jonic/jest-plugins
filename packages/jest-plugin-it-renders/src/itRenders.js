// Libraries
import Renderer from 'react-test-renderer';


const itRenders = (render) => {
  it('renders without error', () => {
    expect(render).not.toThrow();
  });

  it('renders snapshot', () => {
    const rendered = render();
    const element = Renderer.create(rendered).toJSON();

    expect(element).toMatchSnapshot();
  });
};


export default itRenders;
