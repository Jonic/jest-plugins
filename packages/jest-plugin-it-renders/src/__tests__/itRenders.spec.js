// Libraries
import React from 'react';

// Modules
import itRenders from '../itRenders';


// Test Components
const View = () => <div />;
const Text = () => <div />;


/* eslint-disable no-undef */
describe('itRenders', () => {
  describe('View', () => {
    itRenders(() => <View />);
  });

  describe('Text', () => {
    itRenders(() => <Text />);
  });
});
