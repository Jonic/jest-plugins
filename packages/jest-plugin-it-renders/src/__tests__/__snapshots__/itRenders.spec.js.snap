// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itRenders Text renders snapshot 1`] = `<div />`;

exports[`itRenders View renders snapshot 1`] = `<div />`;
